galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager.ivt/src/main/java/dev/galasa/cicsts/manager/ivt/CICSTSManagerIVT.java

```diff
@@ -0,0 +1,199 @@
+   @Test
+   public void testCicsResource() throws CicstsManagerException  {
+	   logger.info("Testing that the CICS TS Manager retrieves a CICS Resource");
+	   ICicsResource cicsResource = cics.cicsResource();
+	   assertThat(cicsResource).isNotNull();
+   }
+   
+   /**
+    * Tests that the CICS TS Manager retrieves the correct job
+    * @throws ZosBatchException 
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testGetRegionJob() throws CicstsManagerException, ZosBatchException, TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS TS Manager finds the DSE CICS job for the CICS Region");
+	   logger.info("There should be an active job as the CICS Region is running");
+	   boolean exceptionThrown = false;
+	   IZosBatchJob regionJob = null;
+	   try {
+		   regionJob = cics.getRegionJob();
+	   } catch (CicstsManagerException e) {
+		   exceptionThrown = true;
+	   }
+	   assertThat(exceptionThrown).isFalse();
+	   assertThat(regionJob).isNotNull();	   
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager retrieves the correct CICS Region
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testGetCicsRegion() throws TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS Terminal gets the correct CICS Region");
+	   String testCicsRegion = terminal.getCicsRegion().toString().replace("CICS Region[", "").replace("]", "");
+	   logger.info("CICS Region retrieved by the ICicsTerminal: " + testCicsRegion);
+	   terminal.type("CEMT INQUIRE").enter().waitForKeyboard();
+	   assertThat(terminal.retrieveScreen().contains(testCicsRegion)).isTrue();
+   }
+   
+   /**
+    * Tests that the CICS Terminal in the CICS TS Manager correctly connects to the CICS Region
+    * @throws CicstsManagerException
+    * @throws TerminalInterruptedException 
+    * @throws NetworkException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testConnectToCicsRegion() throws CicstsManagerException, TerminalInterruptedException, KeyboardLockedException, NetworkException, TimeoutException {
+	   logger.info("Testing that the CICS Terminal connects to the CICS Region");
+	   logger.info("First, need to disconnect from the CICS Region");
+	   terminal.disconnect();
+	   assertThat(terminal.isConnected()).isFalse();
+	   terminal.connectToCicsRegion();
+	   assertThat(terminal.isConnected()).isTrue();
+	   assertThat(terminal.isClearScreen()).isTrue();
```

Comment:
Lets also check that we connected to the same APPLID that we connected to last time

Reply:
I have checked this and added it to this test case.